docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+ability to stop and restart training. In this section, we’ll modify our script for the purpose of
+checkpointing.
+
+..
+   note::Two Types of Continuations (Stopping and Restarting)
+
+   There are two types of continuations (methods for stopping and restarting
+   training): pausing and reactivating training using the WebUI or clicking
+   **Continue Trial** after the experiment completes.
+
+   Each type of continuation has its own behavior. While you always want to preserve
+   the value you are incrementing (the “model weight”), you do not always
+   want to preserve the batch index. When you pause and reactivate you want
+   training to continue from the same batch index. However, when starting a
+   fresh experiment, you want training to start with a fresh batch index.
+
+   To distinguish between the two types of continuations, you can save the
+   trial ID in the checkpoint.
+
+In this step, we’ll run our experiment using the ``model_def_checkpoints.py`` script and its
+accompanying ``checkpoints.yaml`` experiment configuration file.
+
+Step 3.1: Save Checkpoints
+==========================
+
+To save checkpoints, add the ``store_path`` function to your script:
+
+.. note::
+
+   Refer to line 208 in ``model_def_checkpoints.py``
+
+.. code::
+
+   # NEW: Save checkpoint.
+     checkpoint_metadata_dict = {"steps_completed": steps_completed}
+
+     with core_context.checkpoint.store_path(checkpoint_metadata_dict) as (path, storage_id):
+       torch.save(model.state_dict(), path / "checkpoint.pt")
+
+.. image:: /assets/images/model-def-checkpoints-script.png
+
+Step 3.2: Enable Pausing and Resuming an Experiment
+===================================================
+
+To enable pausing an experiment, enable preemption:
+
+.. code::
+
+   # NEW: Detect when the experiment is paused by the WebUI.
+   if core_context.preempt.should_preempt():
+       return
+
+define a load function:
+
+.. code::
+
+   #NEW: Define load_state function for restarting model training from existing checkpoint.
+   def load_state(checkpoint_directory):
+       checkpoint_directory = pathlib.Path(checkpoint_directory)
+       with checkpoint_directory.joinpath(“checkpoint.pt”).open(“rb”) as f:
+           return torch.load(f)
+
+and call the load function when a checkpoint exists and restore the model before resuming training:
+
+.. code::
+
+   # NEW: If checkpoint exists, load it and assign it to model state prior to resuming training.
+   info = det.get_cluster_info() assert info is not None, “this example
+   only runs on-cluster” latest_checkpoint = info.latest_checkpoint if
+   latest_checkpoint is not None:
+       with core_context.checkpoint.restore_path(latest_checkpoint) as path:
+           model = load_state(path)
+
+Step 3.3: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+.. code::
+
+   det e create checkpoints.yaml .
```

Comment:
optional: I like to include the `-f` in the command because:
 1. it's super useful to know about
 2. lots of users don't seem to find out about it, even after a long time of using determined.  So sticking it in when it's not entirely necessary is good advertising I think.

Reply:
edited 

.. code:: bash

   det e create checkpoints.yaml . -f

The ``-f`` option is the short form of ``--follow``.